src/runtime_src/driver/include/ert.h

```diff
@@ -186,6 +186,72 @@ struct ert_configure_cmd {
   uint32_t data[1];
 };
 
+/**
+ * struct ert_configure_sk_cmd: ERT configure soft kernel command format
+ *
+ * @state:           [3-0] current state of a command
+ * @count:           [22-12] number of words in payload (13 DWords)
+ * @opcode:          [27-23] 1, opcode for configure
+ * @type:            [31-27] 0, type of configure
+ *
+ * @start_cuidx:     start index of compute units
+ * @num_cus:         number of compute units in program
+ * @sk_size:         size in byts of soft kernel image
+ * @sk_name:         symbol name of soft kernel
+ * @data[0]:         the low 32 bits of soft kernel image's physical address
+ * @data[1]:         the high 32 bits of soft kernel image's physical address
+ */
+struct ert_configure_sk_cmd {
+  union {
+    struct {
+      uint32_t state:4;          /* [3-0]   */
+      uint32_t unused:8;         /* [11-4]  */
+      uint32_t count:11;         /* [22-12] */
+      uint32_t opcode:5;         /* [27-23] */
+      uint32_t type:4;           /* [31-27] */
+    };
+    uint32_t header;
+  };
+
+  /* payload */
+  uint32_t start_cuidx;
+  uint32_t num_cus;
+  uint32_t sk_size;		/* soft kernel size */
+  uint32_t sk_name[8];		/* soft kernel name */
+
+  uint32_t data[1];
```

Comment:
Should you maybe just expose uint64_t addr, rather than an interpreted payload?  Of course need beware of little or big endian byte order.

Reply:
Done.